From a2ed38719ff1b5ba490df83091e5b4acad1beb88 Mon Sep 17 00:00:00 2001
From: Shivananda Hebbar <x0hebbar@ti.com>
Date: Fri, 19 Mar 2010 13:25:00 -0600
Subject: [PATCH 070/135] DSPBRIDGE: Change dw_state type to u8

The channel state can have values 0, 1 or 2. Hence changing the type to u8.

Signed-off-by: Shivananda Hebbar <x0hebbar@ti.com>
---
 arch/arm/plat-omap/include/dspbridge/_chnl_sm.h |    2 +-
 arch/arm/plat-omap/include/dspbridge/chnlpriv.h |    8 ++++----
 drivers/dsp/bridge/wmd/chnl_sm.c                |    2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h b/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
index f8bdc93..5f89fc6 100644
--- a/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
+++ b/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
@@ -142,7 +142,7 @@ struct chnl_object {
 	/* Pointer back to channel manager */
 	struct chnl_mgr *chnl_mgr_obj;
 	u32 chnl_id;		/* Channel id */
-	u32 dw_state;		/* Current channel state */
+	u8 dw_state;		/* Current channel state */
 	u32 chnl_mode;		/* Chnl mode and attributes */
 	/* Chnl I/O completion event (user mode) */
 	bhandle user_event;
diff --git a/arch/arm/plat-omap/include/dspbridge/chnlpriv.h b/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
index 9fe7a74..f58f7d9 100644
--- a/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
+++ b/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
@@ -49,9 +49,9 @@
 #define CHNL_MODEMASK       0x1001
 
 /* Higher level channel states: */
-#define CHNL_STATEREADY     0x0000	/* Channel ready for I/O. */
-#define CHNL_STATECANCEL    0x0001	/* I/O was cancelled. */
-#define CHNL_STATEEOS       0x0002	/* End Of Stream reached. */
+#define CHNL_STATEREADY		0	/* Channel ready for I/O. */
+#define CHNL_STATECANCEL	1	/* I/O was cancelled. */
+#define CHNL_STATEEOS		2	/* End Of Stream reached. */
 
 /* Determine if user supplied an event for this channel: */
 #define CHNL_IS_USER_EVENT(mode)  (mode & CHNL_MODEUSEREVENT)
@@ -78,7 +78,7 @@ struct chnl_info {
 	/*Abstraction of I/O completion event. */
 	struct sync_object *sync_event;
 	u32 dw_mode;		/* Channel mode. */
-	u32 dw_state;		/* Current channel state. */
+	u8 dw_state;		/* Current channel state. */
 	u32 bytes_tx;		/* Total bytes transferred. */
 	u32 cio_cs;		/* Number of IOCs in queue. */
 	u32 cio_reqs;		/* Number of IO Requests in queue. */
diff --git a/drivers/dsp/bridge/wmd/chnl_sm.c b/drivers/dsp/bridge/wmd/chnl_sm.c
index 1c669b5..595981d 100644
--- a/drivers/dsp/bridge/wmd/chnl_sm.c
+++ b/drivers/dsp/bridge/wmd/chnl_sm.c
@@ -99,7 +99,7 @@ dsp_status bridge_chnl_add_io_req(struct chnl_object *chnl_obj, void *pHostBuf,
 	struct chnl_irp *chnl_packet_obj = NULL;
 	struct wmd_dev_context *dev_ctxt;
 	struct dev_object *dev_obj;
-	u32 dw_state;
+	u8 dw_state;
 	bool is_eos;
 	struct chnl_mgr *chnl_mgr_obj = pchnl->chnl_mgr_obj;
 	u8 *host_sys_buf = NULL;
-- 
1.7.0.4

